lib/eslint/flat-eslint.js

```diff
@@ -922,7 +928,7 @@ class FlatESLint {
             throw new Error("'options.filePath' must be a non-empty string or undefined");
         }
 
-        if (typeof warnIgnored !== "boolean") {
+        if (!(typeof warnIgnored === "boolean" || typeof warnIgnored === "undefined")) {
```

Comment:
This might be a bit easier to understand.

```suggestion
        if (typeof warnIgnored !== "boolean" && typeof warnIgnored !== "undefined")) {
```